crates/mcp/src/surfpool/mod.rs

```diff
@@ -7,147 +11,204 @@ use rmcp::{
 };
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
-use set_token_account::{SeededAccount, SetTokenAccountResponse, SetTokenAccountsResponse};
+use set_token_account::{SeededAccount, SetAccountSuccess, SetTokenAccountsResponse};
 use start_surfnet::StartSurfnetResponse;
 
 mod set_token_account;
 mod start_surfnet;
 
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone)]
 pub struct Surfpool {
-    pub surfnets: Arc<RwLock<Vec<u16>>>,
+    pub surfnets: Arc<RwLock<HashMap<u16, u16>>>,
 }
 
 impl Surfpool {
     pub fn new() -> Self {
         Self {
-            surfnets: Arc::new(RwLock::new(vec![])),
+            surfnets: Arc::new(RwLock::new(HashMap::new())),
         }
     }
 }
 
+impl Default for Surfpool {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 #[derive(Debug, Clone, Default, Serialize, Deserialize, JsonSchema)]
 pub struct CreateTokenAccountForOwnerParams {
     #[schemars(
-        description = "An optional owner address for the accounts. If provided, all token accounts will be created under this owner. If omitted, a new wallet will be generated and returned."
+        description = "The owner address for the token accounts. If omitted, a new wallet will be generated and its address returned."
     )]
-    pub owner: Option<String>, // Optional owner address for the token account
+    pub owner: Option<String>,
     #[schemars(
-        description = "A list of token parameters to dictate what the owner should be funded with. Each parameter includes the token mint, program ID, and amount."
+        description = "A list of parameters for the tokens to be funded, including mint, program ID, and amount."
     )]
-    pub params: Vec<CreateTokenAccountParams>, // Parameters for creating the token account
+    pub params: Vec<CreateTokenAccountParams>,
 }
 
 #[derive(Debug, Clone, Default, Serialize, Deserialize, JsonSchema)]
 pub struct CreateTokenAccountParams {
+    #[schemars(description = "The base58-encoded mint address of the token to fund.")]
+    pub token_mint: String,
```

Comment:
I think we'll need this to be an `Option<String>` - the consumer can provide a symbol instead of a mint